src/context.c

```diff
@@ -183,6 +310,11 @@ void r_context_prepare(void) {
 	}
 
 	context->pending = FALSE;
+
+	if (!determine_slot_states(&error)) {
+		g_warning("Could not determine slot states: %s", error->message);
+		g_clear_error(&error);
+	}
```

Comment:
You are completely right: a warning is not sufficient here.
Deeper analysis with today's experience suggests going a different way. I will push a refactored series soon.